hedera-node/src/main/java/com/hedera/services/txns/crypto/AutoCreationLogic.java

```diff
@@ -165,28 +168,37 @@ public void submitRecordsTo(final RecordsHistorian recordsHistorian) {
      *
      * @param change a triggering change with unique alias
      * @param accountsLedger the accounts ledger to use for the provisional creation
+     * @param changes list of all changes need to construct tokenAliasMap
      * @return the fee charged for the auto-creation if ok, a failure reason otherwise
      */
     public Pair<ResponseCodeEnum, Long> create(
             final BalanceChange change,
-            final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger) {
+            final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger,
+            final List<BalanceChange> changes) {
         if (!usageLimits.areCreatableAccounts(1)) {
             return Pair.of(MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED, 0L);
         }
-        final var alias = change.alias();
+
+        final ByteString alias = getAlias(change);
+
+        if (shouldConstructTokenAliasMap(change)) {
+            // constructs tokenAliasMap lazily if the change consists of any token transfer to
+            // unknown alias. This map is used to count number of maxAutoAssociations needed on
+            // auto created account
+            analyzeTokenTransferCreations(changes);
+        }
+
+        final var maxAutoAssociations = change.isForHbar() ? 0 : tokenAliasMap.get(alias).size();
```

Comment:
Is it not possible for an auto-created account to receive both hbar and tokens in the same `CryptoTransfer`? (I.e., could an account initially auto-created from an hbar credit _also_ get tokens it needs auto-associations with?)

Reply:
I don't think so. Because then it considers the account is already created , but it doesn't have `maxAutoAssociations` set on it. So it returns `TOKEN_NOT_ASSOCIATED_TO_ACCOUNT`. Do you think that should work as well ?

Reply:
Changed to work with both hbar and tokens in a single transfer